library/src/device/generator/stockham_pp_gen_cc.h

```diff
@@ -0,0 +1,998 @@
+// Copyright (C) 2025 Advanced Micro Devices, Inc. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#pragma once
+#include "stockham_gen_cc.h"
+
+// TODO: Once partial pass is fully configurable in kernel-generator.py:
+//      - Test with factors_pp.size() > 1.
+//      - Revisit all usages of transforms_per_block_pp and threads_per_transform.
+//      - Different input/output strides.
+//      - Revisit mod 128 usage in calculate_offsets() with different input lengths,
+//        (logic is required to work with nbatch > 1)
+//      - Revisit factor 192 logic in calculate_offsets() with different input lengths
+//      - Revisit and test local transpose logic for different input lengths
+
```

Comment:
I would really appreciate some documentation here; it's hard to tell why these functions are needed, and how they fit into the larger picture,

Reply:
I've added some comments for both generators, and that should hopefully clarify their purposes.